recipes-bsp/u-boot/files/0028-board-siemens-iot2050-implement-the-load-and-configu.patch

```diff
@@ -0,0 +1,342 @@
++	}
++}
++
++static void
++get_serdes_mux_ctrl(struct serdes_mux_control_pin *serdes_mux_ctrl_info, int m2_connector_type)
++{
++	switch (m2_connector_type) {
++	case BKEY_PCIEX2_EKEY_NONE:
++		*serdes_mux_ctrl_info = m2_config_table[0].serdes_mux_ctrl_pin;
++		break;
++	case BKEY_PCIE_EKEY_PCIE:
++		*serdes_mux_ctrl_info = m2_config_table[1].serdes_mux_ctrl_pin;
++		break;
++	case BKEY_USB30_EKEY_PCIE:
++		*serdes_mux_ctrl_info = m2_config_table[5].serdes_mux_ctrl_pin;
++		break;
++	default:
++		*serdes_mux_ctrl_info = m2_config_table[5].serdes_mux_ctrl_pin;
++		break;
++	}
++}
++
++static void configure_serdes_mux(struct serdes_combination serdes_selected_combination)
++{
++	struct serdes_mux_control_pin serdes_mux_ctrl_info;
++
++	get_serdes_mux_ctrl(&serdes_mux_ctrl_info, serdes_selected_combination.m2_connector_type);
++
++	set_pinvalue(serdes_mux_ctl_pin_info[0].gpio_name, serdes_mux_ctl_pin_info[0].label,
++				serdes_mux_ctrl_info.ctrl_usb30_pcie0_lane0);
++	set_pinvalue(serdes_mux_ctl_pin_info[1].gpio_name, serdes_mux_ctl_pin_info[1].label,
++				serdes_mux_ctrl_info.ctrl_pcie1_pcie0);
++	set_pinvalue(serdes_mux_ctl_pin_info[2].gpio_name, serdes_mux_ctl_pin_info[2].label,
++				serdes_mux_ctrl_info.ctrl_usb30_pcie0_lane1);
++}
++
++
++static void set_m2_load_info(const char **fdt_name)
++{
++	int i;
++	int config_table_size = ARRAY_SIZE(m2_config_table);
++	struct m2_device_config_decodes	config;
++	struct serdes_combination current_serdes_combination =
++												{BKEY_USB30_EKEY_PCIE, USB30, PCIE1_LANE0};
++	enum m2_connector_combination_type current_m2_connector_type;
++
++	if (env_get("m2_manual_config")) {
++		printf("Maunal Select M.2 Configuration\n");
++		current_m2_connector_type = simple_strtoul(env_get("m2_manual_config"), NULL, 10);
++		get_serdes_combination(&current_serdes_combination, current_m2_connector_type);
++	} else { /*auto detect*/
++		get_m2_config_decodes(&config);
++		for (i = 0; i < config_table_size; i++) {
++			if (!memcmp(&config, &m2_config_table[i].m2_config_decodes,
++						sizeof(struct m2_device_config_decodes))) {
++				current_serdes_combination =  m2_config_table[i].serdes_select;
++				break;
++			}
++		}
++	}
++
++	update_m2_load_info(current_serdes_combination, fdt_name);
++	configure_serdes_mux(current_serdes_combination);
++}
++
++/*
++ * M.2 Board is a Variant Baesed On PG2 board
++ */
++static bool board_is_m2(void)
++{
++	struct iot2050_info *info = IOT2050_INFO_DATA;
++
++	return info->magic == IOT2050_INFO_MAGIC &&
++		strcmp((char *)info->name, "IOT2050-ADVANCED-M2") == 0;
++}
++
++static void m2_power_enable(void)
++{
++	/* enable m.2 connector power */
++	set_pinvalue(m2_enable_power.gpio_name, m2_enable_power.label, 1);
++	udelay(4 * 100);
++}
++
+ static bool board_is_advanced(void)
+ {
+ 	struct iot2050_info *info = IOT2050_INFO_DATA;
+@@ -79,7 +318,7 @@ void set_board_info_env(void)
+ 		return;
+ 	}
+ 
+-	if (env_get("board_uuid"))
++	if (env_get("board_uuid") && !board_is_m2())
```

Comment:
NACK, we must not run the block below on every boot, just because it's an M.2

Reply:
put  the m2 operation into the fixup block to avoid the modification.